### ISSUE TITLE
libwabt: Remove from xplat release builds

### ISSUE THREAD

**Issue by obastemur** [Microsoft/ChakraCore#3789]
Offline with @Cellule and learned`libwabt` shouldn't be part of xplat release builds.

Opening this issue to track. Assigning myself && @Cellule . One of us should take care of this.

**Comment by Cellule** [comment]
In CommonDefines.h we have the following
```cpp
#if defined(_WIN32) || (defined(__clang__) && defined(ENABLE_DEBUG_CONFIG_OPTIONS))
#define ENABLE_WASM

#ifdef CAN_BUILD_WABT
#define ENABLE_WABT // This won't be set on clang build if ENABLE_DEBUG_CONFIG_OPTIONS is not set
#endif

#endif
#endif
```

Which means there should at least be no way to access wabt's code because it will be `#define` out, well the ChakraCore part that calls into wabt will be missing. I don't know if the actual code for wabt would still be in there.

What we should do is change `root/CMakeLists.txt`
```cmake
if(CC_TARGETS_AMD64)
    add_definitions(-D_M_X64_OR_ARM64)
    add_compile_options(-msse4.2)
# Do not set this on release build, not sure how to check for that
    set(CAN_BUILD_WABT 1)
```

**Comment by obastemur** [comment]
@Cellule Thanks for details.

> > Which means there should at least be no way to access wabt's code because it will be #define out....

Indeed but we force load the libs hence it's part of the binary release :/

#3788 should take care of this.

**Comment by obastemur** [comment]
Update is already merged (both 1.7 and master)
